activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,21 @@
+*   Change `remove_columns` to use a single query on supported adapters.
+
+    The MySQL and PostgreSQL adapters support bulk `ALTER TABLE` statements.
+
+    ```ruby
+    remove_columns :my_table, :col_one, :col_two
+    ```
+
+    Now results in the following SQL:
+
+    ```sql
+    ALTER TABLE "my_table" DROP COLUMN "col_one", DROP COLUMN "col_two"
+    ```
+
+    SQLite3 does not support bulk `ALTER TABLE` statements.
```

Comment:
The SQLite adapter does benefit from this patch - it also now drops all columns in a single operation instead of one at a time, although in its case the operation is a convoluted new table + copy + rename dance, rather than a single ALTER statement. No longer copying the table N times to drop N columns is still a win, though.

I think we should avoid mentioning adapter support at all, rather than try to explain the different SQLite behaviour, since as far as I can tell the existing workarounds aren't publicly documented anywhere.

```suggestion
*   Optimize `remove_columns` to use a single SQL statement.

    ```ruby
    remove_columns :my_table, :col_one, :col_two
    ```

    Now results in the following SQL:

    ```sql
    ALTER TABLE "my_table" DROP COLUMN "col_one", DROP COLUMN "col_two"
    ```
```